### ISSUE TITLE
ResourceNotFound in ZipReadFS

### ISSUE THREAD

**Issue by dAnjou** [althonos/fs.archive#6]
Hi,

the following code:
```python3
from io import BytesIO
from typing import Any

import fs
import requests
from fs.archive.tarfs import TarReadFS
from fs.archive.zipfs import ZipReadFS
from fs.tarfs import ReadTarFS
from fs.zipfs import ReadZipFS


def print_readme(klass: Any, url: str) -> None:
    data = requests.get(url).content
    fs = klass(BytesIO(data))
    sub_fs = fs.opendir(fs.listdir("/")[0])
    print(sub_fs.listdir("."))
    print(sub_fs.open("README.md").read().strip())


if __name__ == '__main__':
    print(f"VERSION: {fs.archive.__version__}\n")

    data = [
        (TarReadFS, "https://gitlab.com/dAnjou/test/-/archive/main/test-main.tar.gz"),
        (ReadTarFS, "https://gitlab.com/dAnjou/test/-/archive/main/test-main.tar.gz"),
        (ZipReadFS, "https://gitlab.com/dAnjou/test/-/archive/main/test-main.zip"),
        (ReadZipFS, "https://gitlab.com/dAnjou/test/-/archive/main/test-main.zip"),
    ]

    for klass, url in data:
        print(f"### BEGIN {klass}")
        try:
            print_readme(klass, url)
        except Exception as e:
            print(e)
        print("### END\n")
```
produces this output:
```
VERSION: 0.7.0

### BEGIN <class 'fs.archive.tarfs.TarReadFS'>
['README.md']
# Test
### END

### BEGIN <class 'fs.tarfs.ReadTarFS'>
['README.md']
# Test
### END

### BEGIN <class 'fs.archive.zipfs.ZipReadFS'>
['README.md']
resource 'README.md' not found
### END

### BEGIN <class 'fs.zipfs.ReadZipFS'>
['README.md']
# Test
### END
```
Could you have a look, please? 🙂

**Comment by dcoshea** [comment]
It looks like `open()` fails for any filename that contains uppercase characters, but succeeds if the filename is entirely lowercase:

```
>>> zip.listdir('/testdir')
['lower', 'Mixed', 'UPPER']
>>> zip.open('/testdir/lower')
<_io.TextIOWrapper name='/testdir/lower' encoding='utf-8'>
>>> zip.open('/testdir/Mixed')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "[...]/site-packages/fs/base.py", line 1224, in open
    bin_file = self.openbin(path, mode=bin_mode, buffering=buffering)
  File "[...]/site-packages/fs/archive/zipfs/__init__.py", line 213, in openbin
    raise errors.ResourceNotFound(path)
fs.errors.ResourceNotFound: resource '/testdir/Mixed' not found
>>> zip.open('/testdir/UPPER')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "[...]/site-packages/fs/base.py", line 1224, in open
    bin_file = self.openbin(path, mode=bin_mode, buffering=buffering)
  File "[...]/site-packages/fs/archive/zipfs/__init__.py", line 213, in openbin
    raise errors.ResourceNotFound(path)
fs.errors.ResourceNotFound: resource '/testdir/UPPER' not found
```

I didn't check what happens if the directory name includes uppercase.

Edit: This was seen with:
```
fs==2.4.15
fs.archive==0.7.1
```

**Comment by althonos** [comment]
Found the bug, this was caused by a `str.lower` call that shouldn't have been there.

**Comment by althonos** [comment]
Should be fixed in `v0.7.2`.

**Comment by krateng** [comment]
This still happens for me on 0.7.3 for folders:

```console
touch folder/file folder/OtherFile OtherFolder/file OtherFolder/OtherFile
zip test.zip */*
rm -r folder OtherFolder
```

```python
import fs
from fs.archive.zipfs import ZipReadFS

localdirfs = fs.open_fs('.')
archivefs = ZipReadFS('test.zip')

fs.copy.copy_dir(src_fs=archivefs,src_path="/folder",dst_fs=localdirfs,dst_path="/folder") # succeeds
fs.copy.copy_dir(src_fs=archivefs,src_path="/OtherFolder",dst_fs=localdirfs,dst_path="/OtherFolder") #fails
```

```
  File "/home/krateng/.local/lib/python3.10/site-packages/fs/copy.py", line 347, in copy_dir
    copy_dir_if(
  File "/home/krateng/.local/lib/python3.10/site-packages/fs/copy.py", line 436, in copy_dir_if
    copy_structure(src_fs, dst_fs, walker, src_path, dst_path)
  File "/home/krateng/.local/lib/python3.10/site-packages/fs/copy.py", line 311, in copy_structure
    for dir_path in walker.dirs(_src_fs, src_root):
  File "/home/krateng/.local/lib/python3.10/site-packages/fs/walk.py", line 375, in dirs
    for _path, info in self._iter_walk(fs, path=path):
  File "/home/krateng/.local/lib/python3.10/site-packages/fs/walk.py", line 427, in _walk_breadth
    for info in _scan(fs, dir_path, namespaces=namespaces):
  File "/home/krateng/.local/lib/python3.10/site-packages/fs/walk.py", line 287, in _scan
    for info in fs.scandir(dir_path, namespaces=namespaces):
  File "/home/krateng/.local/lib/python3.10/site-packages/fs/archive/zipfs/__init__.py", line 181, in scandir
    raise errors.ResourceNotFound(path)
fs.errors.ResourceNotFound: resource '/OtherFolder' not found
```
